### PR TITLE
feat(semantic-zk-ci): configured full-featured semantic CI w/t linting

### PR DESCRIPTION
**Pull request by keriat**
* github action flows aligned with the general templates
 * husky upgraded & configured
 * cspell upgraded & configured, `cspell-zksync.txt` dict updated
 * dependencies fixed
 * markdownlint upgraded & configured

### CODE REVIEW

File: .github/workflows/deploy-preview.yml [hunk 1]
```diff
@@ -10,23 +7,23 @@ jobs:
...
 
-      - name: "Prepare: yarn@3 & firebase-tools setup"
+      - name: "Install dependencies"
         run: |
           yarn set version 3.1.1
           yarn install
-          npm install -g firebase-tools
-      - name: "Build: dapp"
+
+      - name: "Generate rinkeby dist"
```
Comment (popzxc) [inline .github/workflows/deploy-preview.yml]: Why "rinkeby" exactly?

File: .github/workflows/deploy-preview.yml [hunk 2]
```diff
@@ -10,23 +7,23 @@ jobs:
...
         run: yarn run zk-ci-prepare
 
-      - name: "Deploy: preview"
+      - name: "Deploy preview: ${{ github.event.inputs.network || 'rinkeby' }}"
```
Comment (popzxc) [inline .github/workflows/deploy-preview.yml]: Is the network relevant to the docs?

File: cspell-zksync.txt [hunk 3]
```diff
@@ -72,15 +67,35 @@ subsecond
 supermajority
 gemeente
 unauthorised
-Ethereum's
 SDKs
+inter alia
+unbundle
+incentivized
+authentification
```
Comment (popzxc) [inline cspell-zksync.txt]: This word is incorrect, should not be on the list.

**Review by popzxc — APPROVED** [review]
Nice, just a few nits.

**Review by barakshani — APPROVED** [review]

**Review by StanislavBreadless — APPROVED** [review]

**Pull request merged**